tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -30,6 +44,183 @@
+                         " for {}".format(self.logical_intf))
+            disable_dom_result = self.duthost.command(self.cmd_disable_dom)
+            assert disable_dom_result["rc"] == 0, \
+                "Disable DOM polling failed for {}".format(self.logical_intf)
+            time.sleep(self.wait_after_dom_config)
+
+        # It's needed to shutdown ports before reset and startup ports after reset,
+        # to get config/state machine/etc replayed, so that the modules can be fully
+        # restored.
+        logging.info("Shutdown {} before sfp reset".format(self.logical_intf))
+        shutdown_result = self.duthost.command(self.cmd_down)
+        assert shutdown_result["rc"] == 0, "Shutdown {} failed".format(self.logical_intf)
+        assert check_interface_status(self.duthost, [self.logical_intf], expect_up=False)
+
+        if not self.skip_lpmode_handle:
+            logging.info("Check output of '{}' before sfp reset".format(self.cmd_sfp_show_lpmode))
+            lpmode_show = self.duthost.command(self.cmd_sfp_show_lpmode)
+            self.original_lpmode_state = \
+                parse_output(lpmode_show["stdout_lines"][2:])[self.logical_intf].lower()
+
+    def restore(self):
+        """
+        Restore a logical interface from disabled state by doing below:
+            * Check and restore lpmode if needed
+            * Startup port
+            * Enable DOM polling
+        """
+        if not self.skip_lpmode_handle:
+            # Restore lpmode if needed, because sfp reset will reset lpmode to default
+            logging.info("Check output of '{}' after sfp reset".format(self.cmd_sfp_show_lpmode))
+            lpmode_show = self.duthost.command(self.cmd_sfp_show_lpmode)
+            after_lpmode_state = \
+                parse_output(lpmode_show["stdout_lines"][2:])[self.logical_intf].lower()
+            if after_lpmode_state != self.original_lpmode_state:
+                logging.info("Restoring {} physical interface {} to lpmode {}".format(
+                    self.logical_intf, self.phy_intf, self.original_lpmode_state))
+                cmd_sfp_set_lpmode_to_original = "{} {} {}".format(cmd_sfp_set_lpmode,
+                                                                   self.original_lpmode_state,
+                                                                   self.logical_intf)
+                lpmode_set_result = self.duthost.command(cmd_sfp_set_lpmode_to_original)
+                assert lpmode_set_result["rc"] == 0, \
+                    "'{}' failed".format(cmd_sfp_set_lpmode_to_original)
+                time.sleep(self.wait_after_lpmode_set)
+
+        logging.info("Startup {} after sfp reset to restore module".format(self.logical_intf))
+        startup_result = self.duthost.command(self.cmd_up)
+        assert startup_result["rc"] == 0, "Startup {} failed".format(self.logical_intf)
+        assert check_interface_status(self.duthost, [self.logical_intf], expect_up=True)
```

Comment:
do we need a per port check here? I see there is a overall check at line 446

Reply:
per-port check is to make sure each module is in expected state at each step when testing the current port. 
overall check is to double check just in case, e.g. if any not-currently-being-tested port gets impacted and goes down as collateral.